addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3785,8 +3785,12 @@ const SnippetOptionWidget = Widget.extend({
                 // Disable all transitions for the duration of the style check
                 // as we want to know the final value of a property to properly
                 // update the UI.
+                this.options.wysiwyg.odooEditor.observerUnactive();
                 this.$target[0].classList.add('o_we_force_no_transition');
-                const _restoreTransitions = () => this.$target[0].classList.remove('o_we_force_no_transition');
+                const _restoreTransitions = () => {
+                    this.$target[0].classList.remove("o_we_force_no_transition");
+                    this.options.wysiwyg.odooEditor.observerActive();
+                };
```

Comment:
Mmh, indeed. Maybe we could do something more generic and just disable the observer during the call to _computeWidgetState (whatever the function is), no? And it should probably in 14.0/15.0 too?

Reply:
@qsm-odoo , here is the PR (https://github.com/odoo/odoo/pull/127606) for the fix in 15.0 (note that the problem is not there in 14.0 as the logic that handles the `o_dirty` class was a bit different). I removed the fix on this PR so the runbot will be red (as the test introduced in this PR does not benefit from all the fixes) but a soon as the other one is merged, I will rebase this one :+1: